### PR TITLE
Debt - 5133 - Remove roles salaries

### LINKED ISSUE #5133: Hide, remove, deprecate, delete, destroy the "Role and salary expectations" section of the profile
The roles and salary expectations page is not currently in use and even the intended future use will now be replaced by the "Ongoing Recruitment" feature. So now we can safely remove it.

At a minimum we could simply hide the section on the profile and remove associated validation for it. We also need to remove/hide the corresponding section on the admin view of the profile and the "print profile", and the CSV user & candidate downloads.

At a maximum we could seek out and destroy all references to this in the code.

### Acceptance Criteria
- [ ] Delete role and salary components / pages
- [ ] Delete routes
- [ ] Fix the completion check
- [ ] Deprecate field in api

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5133 

## 👋 Introduction

This removes the `expectedSalary`, `expectedClassifications` and `expectedGenericJobTitles` from the application and deprecates the fields in the API.

## 🕵️ Details

We no longer use these anywhere in the application so they are now redundant.

> **Note**
> Some references were left for compatibility reasons.

## 🧪 Testing

1. Refresh API `docker-compose run --rm maintenance bash refresh_api.sh`
2. Build frontend
    1. `npm run codegen --workspaces --if-present`
    2. `npm run production --workspace=admin --workspace=talentsearch`
3. Navigate to `/admin/users`
    1. [ ] Confirm classifications are gone from filters
    2. [ ] Confirm role and salary does not exist on view users
    3. [ ] Print a user and confirm expected role and salary does not appear
    4. [ ] Export CSV and confirm expected role and salary does not appear
5. Navigate to `/talent/profile`
    1. [ ] Confirm role and salary section does not appear
    2. [ ] Confirm you can still complete your profile
6. Create an application
    1. [ ] Confirm role and salary section does not appear
    2. [ ] Confirm you can complete the application with no role and salary expectations

### CODE REVIEW

**Comment by mnigh** [comment]
@tristan-orourke is this meant to be reviewed or should it be moved into the product backlog?

**Comment by tristan-orourke** [comment]
@mnigh This should be reviewed, unless there's something I'm missing.

**Comment by mnigh** [comment]
> @mnigh This should be reviewed, unless there's something I'm missing.

@tristan-orourke I wasn't sure, based on the discussion with Gray earlier today about possibly not doing this.

**Comment by patcon** [comment]
gray: decision was to merge, but then not deploy until ready.

**Review by petertgiles — APPROVED** [review]
Looks good to me!  I found a few bugs but they preexisted this branch. :smile:

File: frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx [hunk 1]
```diff
@@ -300,14 +299,6 @@ const usePoolCandidateCsvData = (candidates: PoolCandidate[]) => {
         description: "CSV Header, Disabled column",
       }),
     },
-    {
-      key: "expectedClassification",
-      label: intl.formatMessage({
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx:305]: How do I do that? 😅

[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx:305]: Thanks, I thought this was done in the CI. Wasn't aware we had to do it!

[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx:305]: Hmm, that would be a good improvement to the translation check CI.  :thinking: I'll throw something up on the board.

File: api/storage/app/lighthouse-schema.graphql [hunk 2]
```diff
@@ -43,8 +43,8 @@ type Applicant {
   locationPreferences: [WorkRegion]
   locationExemptions: String
   acceptedOperationalRequirements: [OperationalRequirement]
-  expectedSalary: [SalaryRange]
-  expectedClassifications: [Classification]
+  expectedSalary: [SalaryRange] @deprecated(reason: "Expected role\/salary being removed")
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/storage/app/lighthouse-schema.graphql:46]: Created https://github.com/GCTC-NTGC/gc-digital-talent/issues/5290

File: api/tests/Feature/ApplicantTest.php [hunk 3]
```diff
@@ -1660,8 +1556,7 @@ public function testNullFilterEqualsUndefinedPoolCandidate() {
                 'where' => [
                     'applicantFilter' => [
                         'equity' => null,
-                        'expectedClassifications' => null,
-                        'hasDiploma' => null,
+                            'hasDiploma' => null,
```
[the remark this answers is not in the document]
Comment (esizer) [reply api/tests/Feature/ApplicantTest.php]: Fixed in [run php-cs-fixer](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5190/commits/1803c4649264ee803908495c8d09c7f2f0e1e874)

File: frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/poolCandidate/usePoolCandidateCsvData.tsx:305]: Lots of red now lol

[prune translation files](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5190/commits/432bd8ab4704b4da461efa54e18489edd5d56aed)

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Excellent work Eric!

I got a couple more things to remove: [`RoleSalarySection.tsx`](https://github.com/GCTC-NTGC/gc-digital-talent/blob/feature/5125-add-skills-info-pool-candidates-csv/frontend/common/src/components/UserProfile/ProfileSections/RoleSalarySection.tsx#L88)

https://github.com/GCTC-NTGC/gc-digital-talent/blob/feature/5125-add-skills-info-pool-candidates-csv/frontend/common/src/components/UserProfile/UserProfile.stories.tsx#L84-L85

**Comment by esizer** [comment]
> Looks like I was too late. I can remove that last file in my PR if you want?

Oh, so sorry. I merged a little too early. That would be great if you could 😜

**Pull request merged**